From 1026c27ed8ee5c7a1db63ae1db0c15d46f8955d0 Mon Sep 17 00:00:00 2001
From: Colin Guthrie <colin@mageia.org>
Date: Tue, 16 Oct 2012 11:48:35 +0100
Subject: [PATCH 509/510] journal: Properly track the number of allocated
 windows.

Checks were already in place to make sure that the number of
windows was limited to 64, but the count was never incremented
or decremented.
---
 src/journal/mmap-cache.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/journal/mmap-cache.c b/src/journal/mmap-cache.c
index 88fe523..7813f03 100644
--- a/src/journal/mmap-cache.c
+++ b/src/journal/mmap-cache.c
@@ -130,6 +130,7 @@ static void window_free(Window *w) {
         assert(w);
 
         window_unlink(w);
+        w->cache->n_windows--;
         free(w);
 }
 
@@ -157,6 +158,7 @@ static Window *window_add(MMapCache *m) {
                 w = new0(Window, 1);
                 if (!w)
                         return NULL;
+                m->n_windows++;
         } else {
 
                 /* Reuse an existing one */
-- 
1.7.12.3

